data_source_quantum_query_json.go

```
@@ -39,28 +49,34 @@ func dataSourceQuantumQueryJSON() *schema.Resource {
 	}
 }
 
-func dataSourceQuantumQueryJSONRead(d *schema.ResourceData, m interface{}) error {
-	json := d.Get("json").(string)
+func dataSourceQuantumQueryJSONRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	jsonStr := d.Get("json").(string)
 	query := d.Get("query").(string)
-	queryResult := gjson.Get(json, query)
+	queryResult := gjson.Get(jsonStr, query)
 
-	d.SetId(fmt.Sprintf("%d-%d", hashcode.String(json), hashcode.String(query)))
+	d.SetId(fmt.Sprintf("%d-%d", computeHash(jsonStr), computeHash(query)))
 	d.Set("result", queryResult.String())
 
 	if queryResult.IsArray() {
-		resultList := []string{}
+		var resultList []string
 		for _, value := range queryResult.Array() {
 			resultList = append(resultList, value.String())
 		}
-		d.Set("result_list", resultList)
+		if err := d.Set("result_list", resultList); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
 	}
 	if queryResult.IsObject() {
-		resultMap := map[string]string{}
+		resultMap := make(map[string]string)
 		for key, value := range queryResult.Map() {
 			resultMap[key] = value.String()
 		}
-		d.Set("result_map", resultMap)
+		if err := d.Set("result_map", resultMap); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
```

Were we ignoring the error in the past? Or did `Set()` changed and now returns an error?

---

no Set() always returned an error, we simply weren't handling the error case gracefully before. This is better practice